automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/PublishSpec.scala

```diff
@@ -116,5 +115,47 @@ class PublishSpec extends FreeSpec with WebBrowserSpec with UserFixtures with Wo
         }
       }
     }
+
+  }
+
+  "As a user with no TCGA permissions" - {
+    "a published workspace" - {
+      "with TCGA control access" - {
+        "should see a information message about getting TCGA access" in withWebDriver { implicit driver =>
+          //log in as a Curator and create/publish TCGA workspace
+          val curatorUser = UserPool.chooseCurator
+          implicit val curatorAuthToken: AuthToken = curatorUser.makeAuthToken()
+
+          api.NIH.addUserToNIH(Config.Users.jwt)
+          withWorkspace(namespace, "TCGA_", Set(Config.FireCloud.tcgaAuthDomain)) { wsName =>
+            withCleanUp {
+              val data = LibraryData.metadata + ("library:datasetName" -> wsName)
+              api.library.setLibraryAttributes(namespace, wsName, data)
+              register cleanUp api.library.unpublishWorkspace(namespace, wsName)
+              api.library.publishWorkspace(namespace, wsName)
+
+              //log in as a user with no TCGA access to make sure TCGA info message is displayed to you in Libraryval studentUser = UserPool.chooseStudent
+              val studentUser = UserPool.chooseStudent
+              withSignIn(studentUser) { _ =>
+                val page = new DataLibraryPage().open
+                page.hasDataset(wsName) shouldBe true
+                page.openDataset(wsName)
+                //verify that Request Access modal is shown
+                val requestAccessModal   = RequestAccessModal()
+                requestAccessModal.validateLocation shouldBe true
+                requestAccessModal.getRequestAccessText.contains(requestAccessModal.requestAccessText)
+                requestAccessModal.clickOk()
```

Comment:
Is this right? I thought it just had a "Done" button which, at least in this case, maps to `clickCancel()` since it's simply dismissing the modal without doing anything.

Reply:
looks like ok-button is right

Reply:
this particular modal only has OK button